src/jenkins_job_insight/main.py

```
@@ -619,6 +620,73 @@ async def dispatch(self, request: Request, call_next):
+) -> JSONResponse:
+    """Log 422 validation error details at DEBUG level, then return standard response."""
+    from fastapi.encoders import jsonable_encoder
+
+    if logger.isEnabledFor(logging.DEBUG):
+        from jenkins_job_insight.utils import mask_sensitive_fields
+
+        masked_body = None
+        if exc.body is not None:
+            try:
+                masked_body = mask_sensitive_fields(
+                    exc.body if isinstance(exc.body, (dict, list)) else exc.body
```

Ruff (0.15.10)

[warning] Useless `if`-`else` condition

(RUF034)